src/lib/scheduler/node_queue_scheduler.hpp

```diff
@@ -12,63 +12,54 @@ namespace hyrise {
 /*
  * GENERAL SCHEDULING CONCEPT
  *
- * Everything that needs to be processed is encapsulated in tasks. For example, in the context of the database
- * the OperatorTasks encapsulates database operators (here, it only encapsulates the execute function).
- * A task will then be pushed by a Scheduler into a TaskQueue and pulled out by a Worker to be processed.
+ * Everything that needs to be processed is encapsulated in tasks. For example, in the context of the database, the
+ * OperatorTasks encapsulates database operators (here, it only encapsulates the execute function). A task will be
+ * pushed by a Scheduler into a TaskQueue and pulled out by a Worker to be processed.
  *
  *
  * TASK DEPENDENCIES
  *
- * Tasks can be dependent of each other. For example, in the context of the database, a table scan operation can be
- * dependent on a GetTable operation and so do the tasks that encapsulates these operations.
- * Therefore, not every task that is on top of the TaskQueue is ready to be processed. Pulling this unready task and
- * pushing it back to the queue could result in high latency for this task and in the context of databases in a
- * high latency for queries. Not pulling this unready task and waiting until it is ready would block the queue.
- * To avoid these situations, a TaskQueue is implemented as a vector and will be iterated until a ready task is found.
- * This preserves the desired order and will not block the queue.
- * To determine if a task is ready, it checks its parent tasks if they are done. A task will be set done after it
- * was processed successfully.
+ * Tasks can be dependent of each other. For example, a table scan operation can be dependent on a GetTable operation
+ * and so do the tasks that encapsulates these operations. Tasks with predecessors are not scheduled (i.e., added to
```

Comment:
`do the tasks that encapsulate these operations` OR `does the task that encapsulates this operation`

Reply:
The sentence `Tasks with predecessors ...` is not correct, is it? The point is that they are not scheduled if they have unfinished predecessors, right?